trieste/models/gpflow/sampler.py

```diff
@@ -630,80 +628,92 @@ def __init__(
 
         tf.debugging.assert_positive(num_features)
         self._num_features = num_features
-        self._model = model
-        feature_functions = ResampleableDecoupledFeatureFunctions(self._model, self._num_features)
+        feature_functions = ResampleableDecoupledFeatureFunctions(model, self._num_features)
 
-        super().__init__(self._model, feature_functions)
+        super().__init__(model, feature_functions)
 
-    def _prepare_weight_sampler(self) -> Callable[[int], TensorType]:
+    def _prepare_weight_sampler(self) -> Callable[[int], TensorType]:  # [B] -> [B, F + M, L]
         """
         Prepare the sampler function that provides samples of the feature weights
-        for both the RFF and cannonical feature functions, i.e. we return a function
+        for both the RFF and canonical feature functions, i.e. we return a function
         that takes in a batch size `B` and returns `B` samples for the weights of each of
-        the `L`  RFF features and `N` cannonical features.
+        the `F`  RFF features and `M` canonical features for `L` outputs.
         """
 
+        kernel = self._model.get_kernel()
+        if self._feature_functions.is_multioutput:
+
+            def kernel_K(iv: TensorType) -> TensorType:
+                # mypy is confused and doesn't see that kernel can be of two types
```

Comment:
Did you ever find a related mypy issue? If not it might be worth spending a little bit of time (but not too long) creating a minimal example and raising one?

Reply:
I didn't look for a mypy issue. I'll search for one.